src/Helpers/Hooks.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Give\Helpers;
+
+use BadMethodCallException;
+
+class Hooks {
+	/**
+	 * A function which extends the WordPress add_action method to handle the instantiation of a class
+	 * once the action is fired. This prevents the need to instantiate a class before adding it to hook.
+	 *
+	 * @since 2.8.0
+	 *
+	 * @param string $tag
+	 * @param string $class
+	 * @param string $method
+	 * @param int    $priority
+	 * @param int    $acceptedArgs
+	 *
+	 * @return void
+	 */
+	public static function addAction( $tag, $class, $method = '__invoke', $priority = 10, $acceptedArgs = 1 ) {
+		add_action(
+			$tag,
+			static function () use ( $tag, $class, $method ) {
+				// Provide a way of disabling the hook
+				if ( apply_filters( "give_disable_hook-{$tag}", false ) || apply_filters( "give_disable_hook-{$tag}:{$class}@{$method}", false ) ) {
+					return;
+				}
+
+				$instance = give( $class );
+
+				if ( ! method_exists( $instance, $method ) ) {
```

Comment:
`$class` can be passed directly to [method_exists()](https://www.php.net/manual/en/function.method-exists.php) which further defers the object instantiation.

```
mixed $object An object instance or a class name 
```

Reply:
Yes! Great observation!